Cargo.toml

```diff
@@ -13,13 +13,24 @@ keywords = ["integer", "u256", "ethereum"]
 categories = ["cryptography::cryptocurrencies", "mathematics", "no-std"]
 
 [workspace]
-members = [
-  "bench",
-  "intrinsics",
-]
+members = ["bench", "intrinsics"]
 
 [features]
 llvm-intrinsics = ["ethnum-intrinsics"]
+serde = ["impl-serde", "crate-serde"]
+scale = ["parity-scale-codec", "derive_more"]
+default = ["serde"]
 
 [dependencies]
 ethnum-intrinsics = { version = "1", path = "intrinsics", optional = true }
+impl-serde = { version = "0.3", default-features = false, optional = true }
```

Comment:
I looked into the `impl-serde` code and I don't understand why we need to pull in this dependency. `U256::from_str_radix(16)` already provides much of the logic for deserializing, and `impl std::fmt::Hex for U256` provides the serialization logic.

Reply:
Done, although serialization logic still won't reuse std::fmt::Hex, because it causes unnecessary allocations and requires std.

Reply:
I think you can get around allocations and requiring standard by using `core::fmt` facilities.

Specifically, you would have to write your own `core::fmt::Write` implementation for writing formatted data onto some buffer on the stack, but that should be fairly straightforward:

```rust
use core::{fmt::{self, Write}, str};

struct Buffer([u8; 80], usize);

impl Buffer {
    fn new() -> Self {
        Self([0; 80], 0)
    }
    
    fn as_str(&self) -> &str {
        unsafe { str::from_utf8_unchecked(&self.0[..self.1]) }
    }
}

impl Write for Buffer {
    fn write_str(&mut self, s: &str) -> fmt::Result {
        let bytes = self.0[self.1..].get_mut(..s.len()).ok_or(fmt::Error)?;
        bytes.copy_from_slice(s.as_bytes());
        self.1 += s.len();
        Ok(())
    }
}

#[test]
fn writes_buffer() {
    let mut buffer = Buffer::new();
    write!(&mut buffer, "{}", 42).unwrap();
    assert_eq!(buffer.as_str(), "42");
}
```

Reply:
My main motivation for suggesting this is avoiding logic in more than one place for encoding integers as hex.